server/src/api/pay_exec.py

```diff
@@ -30,4 +30,38 @@ def make_payment(self):
         Arguments:
             json -- check api doc for json format
         """
-        return "1"
\ No newline at end of file
+        
+        error_msg = ""
+        success = False
+
+        #extract information from JSON object
+        JSON_object = json.loads(cherrypy.request.body.read().decode('utf-8'))
+        is_paid = JSON_object.get("is_paid")
+        payment_id = JSON_object.get("payment_id")
+        
+        try:
+            #checking if user is logged in successfully
+            username = cherrypy.session["username"]
+            password = cherrypy.session["password"]
+            isLoggedIn = self.database.get_password(username) == password
```

Comment:
If there is a session then the client should be logged in otherwise the code will fail and be caught by the try catch. So, there should be no need to check the password and user name.